5_databases/week1/command.sql

```diff
@@ -0,0 +1,72 @@
+-- 1. Find out how many tasks are in the task table
+
+
+SELECT COUNT(*) 
+FROM task;
+
+-- 2. Find out how many tasks in the task table do not have a valid due date
+
+SELECT id, due_date
+FROM task
+WHERE due_date IS NULL 
+   OR STR_TO_DATE(due_date, '%Y-%m-%d %H:%i:%s') IS NULL;
+
+
+
+-- 3. Find all the tasks that are marked as done
+SELECT status.name, user.id, task.description
+FROM status 
+left JOIN task ON status.id = task.status_id
+LEFT JOIN user ON user.id = task.user_id
+WHERE status.name = 'done';
+
+-- 4. Find all the tasks that are not marked as done
+SELECT status.name, user.id, task.description
+FROM status 
+LEFT JOIN task ON status.id = task.status_id
+left JOIN user ON user.id = task.user_id
+WHERE status.name <> 'done';
+
+-- 5. Get all the tasks, sorted with the most recently created first
+SELECT status.name, created
+FROM task
+JOIN status ON task.status_id = status.id
+LEFT JOIN user ON user.id = task.user_id
+ORDER BY created;
+
+-- 6. Get the single most recently created task
+SELECT status.name, created
+FROM task
+JOIN status ON task.status_id = status.id
+LEFT JOIN user ON user.id = task.user_id
+ORDER BY created DESC LIMIT 1;
+
+-- 7. Get the title and due date of all tasks where the title or description contains database
+SELECT task.title, task.description
```

Comment:
The task asks us to retrieve title and due_date, not description. Well done tho!